ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -520,6 +541,69 @@ public static String getServiceId(RestRequest restRequest) throws RestServiceExc
   }
 
   /**
+   * Gets the isPrivate setting from the args.
+   * @param args The args where to include the isPrivate setting.
+   * @return A boolean to indicate the value of the isPrivate flag.
+   * @throws RestServiceException if exception occurs during parsing the arg.
+   */
+  public static boolean getIsPrivateSetting(Map<String, Object> args) throws RestServiceException {
+    boolean isPrivate;
+    String isPrivateStr = getHeader(args, Headers.PRIVATE, false);
+    if (isPrivateStr == null || isPrivateStr.toLowerCase().equals("false")) {
+      isPrivate = false;
+    } else if (isPrivateStr.toLowerCase().equals("true")) {
+      isPrivate = true;
+    } else {
+      throw new RestServiceException(
+          Headers.PRIVATE + "[" + isPrivateStr + "] has an invalid value (allowed values:true, false)",
+          RestServiceErrorCode.InvalidArgs);
+    }
+    return isPrivate;
+  }
+
+  /**
+   * Gets the name of the target {@link com.github.ambry.account.Account} from a {@link RestRequest}'s header. This
+   * field is expected only for a POST request.
+   * @param restRequest the representation of the request. Cannot be {@code null}.
+   * @return the accountName of the requester's {@link com.github.ambry.account.Account}, or {@code null} if no
+   * accountName was set in the request.
+   * @throws RestServiceException
+   */
+  public static String getTargetAccountNameFromHeader(RestRequest restRequest) throws RestServiceException {
+    Objects.requireNonNull(restRequest, "restRequest cannot be null");
```

Comment:
don't need this - the next line will throw a `NullPointerException` if `restRequest` is `null`